lms/validation/_base.py

```diff
@@ -82,12 +80,15 @@ def parse(self, *args, **kwargs):
 
         :raise lms.validation.ValidationError: if the request isn't valid
         """
-        kwargs.setdefault("locations", self.locations)
-        return self._parser.parse(self, self.context["request"], *args, **kwargs)
+
+        parser = pyramidparser.PyramidParser(
+            location=kwargs.pop("location", self.location),
+            error_handler=self._handle_error,
+        )
+        return parser.parse(self, self.context["request"], *args, **kwargs)
```

Comment:
`locations` used to be an argument to `PyramidParser.parse()`. It's now just `location` and is an argument to `PyramidParser.__init__()` instead. This meant that I had to move the construction of the `PyramidParser` object from a class-level attribute (above) down into this method so `location` could be passed to it

Reply:
Also notice that we're now passing `error_handler=self._handle_error` to `PyramidParser()`, which we weren't before. More on that below